packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -1785,7 +1794,7 @@ class ImportedSubnet extends Resource implements ISubnet, IPublicSubnet, IPrivat
       scope.node.addWarning(`No routeTableId was provided to the subnet ${ref}. Attempting to read its .routeTable.routeTableId will return null/undefined. (More info: https://github.com/aws/aws-cdk/pull/3171)`);
     }
 
-    this.availabilityZone = attrs.availabilityZone;
+    this.availabilityZone = attrs.availabilityZone ?? '';
```

Comment:
Put the potentially undefined value in a private member called `_availabilityZone?: string` and make `availabilityZone` into a getter.